scripts/apps/authoring/authoring/authoring.spec.js

```diff
@@ -66,69 +66,90 @@ describe('authoring', () => {
+                                                        {
+                                                            key: '9i73f',
+                                                            text: 'Annotation 2',
+                                                            type: 'unstyled',
+                                                            depth: 0,
+                                                            inlineStyleRanges: [],
+                                                            entityRanges: [],
+                                                            data: {},
+                                                        },
+                                                        {
+                                                            key: 'd3vb3',
+                                                            text: 'Line 2',
+                                                            type: 'unstyled',
+                                                            depth: 0,
+                                                            inlineStyleRanges: [],
+                                                            entityRanges: [],
+                                                            data: {},
+                                                        },
+                                                    ],
+                                                    entityMap: {},
+                                                }),
+                                                author: 'admin',
+                                                annotationType: 'regular',
+                                            },
+                                        },
                                     },
                                 },
                             },
-                        },
-                    },
-                    inlineStyleRanges: [{
-                        offset: 6,
-                        length: 5,
-                        style: 'ANNOTATION-1',
-                    },
-                    {
-                        offset: 12,
-                        length: 5,
-                        style: 'ANNOTATION-2',
+                            inlineStyleRanges: [{
+                                offset: 6,
+                                length: 5,
+                                style: 'ANNOTATION-1',
+                            },
+                            {
+                                offset: 12,
+                                length: 5,
+                                style: 'ANNOTATION-2',
+                            }],
+                            type: 'unstyled',
+                            text: 'lorem ipsum dolor',
+                        }],
                     }],
-                    type: 'unstyled',
-                    text: 'lorem ipsum dolor',
-                }],
-            }],
+                },
+            },
```

Comment:
I can't comment on line 156 since it hadn't been changed, but I instead of checking `updates.annotations` we need to test if the annotations are in HTML, or we will end up with the same situation as this one where annotations weren't outputted, but the tests were all green.

Reply:
that's tested in [to-html.spec.js](https://github.com/superdesk/superdesk-client-core/blob/master/scripts/core/editor3/html/tests/to-html.spec.js#L89)

Reply:
A test for data needs to be added then because we can still bump into the same problem, that tests pass, but no data is present in the html, only the ids

Reply:
I think ideally it would handle both when processing annotations to generate that html, not sure why it's separate atm, but that would involve a bit of refactoring